From 3f44c3b1d7a528640849b4085e0d7ca9e1e0f7dc Mon Sep 17 00:00:00 2001
From: Joe Bolinger <jbolinger@google.com>
Date: Sat, 5 Apr 2025 02:30:30 +0000
Subject: [PATCH] Drop PendingIntent extras from external packages during
 enrollment.

Bug: 388528350
Flag: EXEMPT bugfix
Test: atest FingerprintEnrollIntroductionTest FaceEnrollIntroductionTest
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:4ccdeee849d5fef78498ba33cadc525523efcbd7)
Merged-In: I61281dcf95e53100a96d6a218f3f00fd1b4ea3f9
Change-Id: I61281dcf95e53100a96d6a218f3f00fd1b4ea3f9
---
 .../biometrics/BiometricEnrollIntroduction.java    | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/com/android/settings/biometrics/BiometricEnrollIntroduction.java b/src/com/android/settings/biometrics/BiometricEnrollIntroduction.java
index e360760184..d1987d6ee5 100644
--- a/src/com/android/settings/biometrics/BiometricEnrollIntroduction.java
+++ b/src/com/android/settings/biometrics/BiometricEnrollIntroduction.java
@@ -47,6 +47,8 @@ import com.google.android.setupdesign.span.LinkSpan;
 import com.google.android.setupdesign.template.RequireScrollMixin;
 import com.google.android.setupdesign.util.DynamicColorPalette;
 
+import java.util.List;
+
 /**
  * Abstract base class for the intro onboarding activity for biometric enrollment.
  */
@@ -224,6 +226,18 @@ public abstract class BiometricEnrollIntroduction extends BiometricEnrollBase
                 });
     }
 
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        if (!getPackageName().equals(getCallingPackage())) {
+            for (String key : List.of(MultiBiometricEnrollHelper.EXTRA_SKIP_PENDING_ENROLL,
+                    MultiBiometricEnrollHelper.EXTRA_ENROLL_AFTER_FACE)) {
+                getIntent().removeExtra(key);
+            }
+        }
+    }
+
     @Override
     protected void onResume() {
         super.onResume();
-- 
2.50.1.351.gcf940e82a1-goog

